packages/shylu/shylu_node/fastilu/src/shylu_fastilu.hpp

```diff
@@ -482,47 +506,75 @@ class FastILUPrec
             lColIdx = OrdinalArray("lColIdx", lRowMap_[nRows]);
             uColIdx = OrdinalArray("uColIdx", uRowMap_[nRows]);
             utColIdx = OrdinalArray("utColIdx", uRowMap_[nRows]);
+
+            lVal = ScalarArray("lVal", lRowMap_[nRows]);
+            uVal = ScalarArray("uVal", uRowMap_[nRows]);
+            utVal = ScalarArray("utVal", uRowMap_[nRows]);
         }
 
         void numericILU()
         {
-            aVal = ScalarArray("aVal", aColIdx.extent(0));
-            aVal_ = Kokkos::create_mirror(aVal);
+            const Scalar zero = STS::zero();
+            #ifdef FASTILU_TIMER
+            Kokkos::Timer Timer;
+            #endif
+            //Sort each row of A by ColIdx
+            KokkosKernels::sort_crs_matrix<ExecSpace, OrdinalArray, OrdinalArray, ScalarArray>(aRowMapIn, aColIdxIn, aValIn);
```

Comment:
Can this be in symbolic?

Reply:
@srajama1, we internally sort the "graph" in symbolic, but when the user inputs the "matrix" for numeric, the matrix is not sorted? We could insert the numerical value, looking for ColIdx match, but sorting the matrix seems to go faster.

Reply:
Understood. Thanks! Typically the structure should not change for numeric, so the position should be the same, but this is ok.